### DISCUSSION TITLE
Edge Function working locally not not when deployed to Vercel

### DISCUSSION THREAD

**Discussion by aabdulahad** [vercel/community#1668] [Help] [answered]
### Summary

I have this edge function, which used to work around 10 days ago. 
I have found this issue (https://github.com/vercel/community/discussions/1618) where someone appears to have a similar problem. I am not sure if it is related.

The code uses the Firebase REST API to convert a Firebase token into their uid to access their document and then update a document. 

A simplified version of the code is below

```js
import { doc, updateDoc } from 'firebase/firestore'
import { NextResponse } from 'next/server'
import { firestore } from '../firebase/firebase'

export const config = {
  runtime: 'edge',
}

export default async function handler(req) {
  try {
    const data = await req.json()

    const userReq = await fetch(
      `https://identitytoolkit.googleapis.com/v1/accounts:lookup?key=${apiKey}`, 
      {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ idToken: data.idToken })
      }
    )
    const user = await userReq.json()
    const uid = user.users[0].localId

    const docRef = doc(firestore, `documents/${uid}`)
    await updateDoc(docRef, {
      // document data
    })
    return NextResponse.json({ message: 'Success', })
  } catch (error) {
    return NextResponse.json({ message: 'Error', })
  }
}
```

### Additional information

The error messages from Vercel logs is either `Error: Network connection lost.`, or Firebase timing out. 
Any help appreciated, thank you.

**Comment by leerob** [comment]
Could you create a reproduction for us to investigate, please? 🙏 A `network connection lost` could also be an intermittent network issue from your internet provider or wifi connection.

**Reply by aabdulahad** [reply]
I have done further testing and isolated the issue to the below code segment:

```js
const docRef = doc(firestore, `documents/${uid}`)
await updateDoc(docRef, {
   // document data
})
```

This produces the error I mentioned in my previous comment - `Uncaught (in promise) SyntaxError: Unexpected token 'A', "An interna"... is not valid JSON` 

It seems like this has some sort of problem with the Edge Runtime, I do not know how to check if the firestore module utilises Node.js APIs and therefore does not work in this environment. Could you give some more advice please.

**Comment by aabdulahad** [comment]
Another error that also appears is: `Uncaught (in promise) SyntaxError: Unexpected token 'A', "An interna"... is not valid JSON`

Of course I can make a minimal reproduction, what is the best method? 
A new minimal code deployment to Vercel or can I use a service like code sandbox?
Thanks

**Comment by adesokanayo** [comment]
I am having this same error from my Next Js Starter template for Langchain.  My app runs locally , but when i deploy to vercel, i get the Error: Network connection lost. Is anyone looking at it ?

**Comment by gxanshu** [comment]
anyone found solution ?

**Comment by 2-fly-4-ai** [comment]
Also getting these errors. "Error connection lost" on edge runtime.

**Reply by tycomo** [reply]
I created an open discussion post at https://github.com/orgs/vercel/discussions/5933 as I am running into the same thing and its only reproducible in production making it hard to debug.

Were you able to resolve these issues?

**Reply by neils94** [reply]
Also receiving the same issue, only in production as well.

**Comment by zineanteoh** [comment]
Was having `EDGE_FUNCTION_INVOCATION_TIMEOUT` error, then did `const maxDuration = 300;` to try to fix it, but now I get `Error: Network connection lost.`. 

Help appreciated. 

EDIT: realized that we don't actually need to use Edge Function, so switched back to Serverless Function -- so this now becomes a non-issue. Would recommend others to consider if it is really a necessity to use Edge.

**Comment by aabdulahad** [comment, marked as answer]
Hi all, check this [post](https://stackoverflow.com/questions/75515845/firebase-auth-with-nextjs13-edge-api-route) from myself on stackoverflow. 
In the end I switched to using serverless functions over the Edge API as I believe that Firebase needs run in the Node runtime. 
Consider this library [`next-firebase-auth-edge` library](https://github.com/awinogrodzki/next-firebase-auth-edge) as is mentioned by jessepinho in the post.
